crates/proof-of-sql/src/sql/proof_exprs/equals_expr.rs

```diff
@@ -102,10 +102,10 @@ impl ProofExpr for EqualsExpr {
     }
 }
 
-#[allow(
-    clippy::missing_panics_doc,
-    reason = "table_length is guaranteed to match lhs.len()"
-)]
+/// Evaluate if the result is zero
+///
+/// # Panics
+/// Panics if `table_length` does not match the length of `lhs`. Shouldn't happen in practice if the code is used as intended.
```

Comment:
Yeah this one is weird. this one seems like it should have a panic doc since there's nothing preventing callers from providing invalid input, even if the only "callers" are internal to proof-of-sql
```suggestion
/// Panics if `table_length` does not match the length of `lhs`.
```